src/python/grpcio/grpc/_server.py

```diff
@@ -259,14 +261,20 @@ def auth_context(self):
                 cygrpc.auth_context(self._rpc_event.call))
         }
 
+    def set_compression(self, compression):
+        with self._state.condition:
+            self._state.compression_algorithm = compression
+
     def send_initial_metadata(self, initial_metadata):
         with self._state.condition:
             if self._state.client is _CANCELLED:
                 _raise_rpc_error(self._state)
             else:
                 if self._state.initial_metadata_allowed:
+                    augmented_metadata = _compression.augment_metadata(
+                        initial_metadata, self._state.compression_algorithm)
```

Comment:
optional: Make `initial_metadata=()` by default, and change this line to `initial_metadata += _compression.create_compression_metadata(self._state.compression_algorithm)`.